src/core/global.c

```diff
@@ -190,6 +179,10 @@ static void nn_global_handler (struct nn_fsm *self,
 static void nn_global_shutdown (struct nn_fsm *self,
     int src, int type, void *srcptr);
 
+/*  Socket holds. */
+static int nn_global_hold_socket(struct nn_sock **sockp, int s);
+static int nn_global_hold_socket_locked(struct nn_sock **sockp, int s);
+static void nn_global_rele_socket(struct nn_sock *);
```

Comment:
Yep, you've got it.  I was going to comment that acquire is usually exclusive, whereas hold is more like a shared lock; although it isn't a lock at all -- its just something to keep the system from tossing the memory out from underneath us while we're using it. :-)